

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>bss.models package &mdash; Bayesian Structured Sparsity 0.0.2 documentation</title>
  

  
  
  
  

  

  
  
    

  

  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="bss.samplers package" href="bss.samplers.html" />
    <link rel="prev" title="bss package" href="bss.html" /> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Bayesian Structured Sparsity
          

          
          </a>

          
            
            
              <div class="version">
                0.0.2
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Modules:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="modules.html">bss</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="bss.html">bss package</a><ul class="current">
<li class="toctree-l3 current"><a class="reference internal" href="bss.html#subpackages">Subpackages</a><ul class="current">
<li class="toctree-l4 current"><a class="current reference internal" href="#">bss.models package</a></li>
<li class="toctree-l4"><a class="reference internal" href="bss.samplers.html">bss.samplers package</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="bss.html#submodules">Submodules</a></li>
<li class="toctree-l3"><a class="reference internal" href="bss.html#module-bss.data">bss.data module</a></li>
<li class="toctree-l3"><a class="reference internal" href="bss.html#module-bss.mvn">bss.mvn module</a></li>
<li class="toctree-l3"><a class="reference internal" href="bss.html#module-bss">Module contents</a></li>
</ul>
</li>
</ul>
</li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Bayesian Structured Sparsity</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
          <li><a href="modules.html">bss</a> &raquo;</li>
        
          <li><a href="bss.html">bss package</a> &raquo;</li>
        
      <li>bss.models package</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/bss.models.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="bss-models-package">
<h1>bss.models package<a class="headerlink" href="#bss-models-package" title="Permalink to this headline">¶</a></h1>
<div class="section" id="submodules">
<h2>Submodules<a class="headerlink" href="#submodules" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="module-bss.models.probit">
<span id="bss-models-probit-module"></span><h2>bss.models.probit module<a class="headerlink" href="#module-bss.models.probit" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="bss.models.probit.Probit">
<em class="property">class </em><code class="descclassname">bss.models.probit.</code><code class="descname">Probit</code><span class="sig-paren">(</span><em>X</em>, <em>Y</em>, <em>R</em>, <em>target_sparsity=0.01</em>, <em>gamma0_v=1.0</em>, <em>lambda_params=(1e-06</em>, <em>1e-06)</em>, <em>nu_params=(1e-06</em>, <em>1e-06)</em>, <em>xi=0.999999</em>, <em>xi_prior_shape=(1</em>, <em>1)</em>, <em>check_finite=True</em>, <em>min_eigenval=0</em>, <em>jitter=1e-06</em><span class="sig-paren">)</span><a class="headerlink" href="#bss.models.probit.Probit" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal"><span class="pre">object</span></code></p>
<p>The Probit model used for modeling Sparse Regression using a Gaussian field. <a class="reference internal" href="index.html#engelhardt2014" id="id1">[EA14]</a>.</p>
<div class="math">
\[y|X,\beta,\beta_0, \nu \propto \mathcal{N}(\beta_0 1_n + X \beta, \nu^{-1} I_n)\]</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>X</strong> : ndarray</p>
<blockquote>
<div><p>The predictor matrix of real numbers, n x p in size, where n is the no. of samples (genotypes) and p is the
no. of features (SNPs).</p>
</div></blockquote>
<p><strong>Y</strong> : ndarray</p>
<blockquote>
<div><p>The response vector of real numbers, n x 1 in size, with each value representing the phenotype value for the
sample.</p>
</div></blockquote>
<p><strong>R</strong> : ndarray</p>
<blockquote>
<div><p>The covariance matrix for the SNPs, p x p in size. The matrix may not be positive-definite, but is converted
to one internally.</p>
</div></blockquote>
<p><strong>target_sparsity</strong> : float</p>
<blockquote>
<div><p>The proportion of included predictors. For example, a value of 0.01 indicates that around 1% of total SNPs
are expected be included in our model. This value affects the probit threshold gamma_0 of the model.</p>
</div></blockquote>
<p><strong>gamma0_v</strong> : float</p>
<blockquote>
<div><p>Variance of the probit threshold gamma_0</p>
</div></blockquote>
<p><strong>lambda_params</strong> : tuple</p>
<blockquote>
<div><p>Shape parameter and Inverse-scale parameter of the gamma prior placed on the model parameter lambda, where
lambda is the inverse squared global scale parameter for the regression weights.</p>
</div></blockquote>
<p><strong>nu_params</strong> : tuple</p>
<blockquote>
<div><p>Shape parameter and Inverse-scale parameter of the gamma prior placed on the model parameter nu, where nu
is the residual precision.</p>
</div></blockquote>
<p><strong>xi</strong> : float</p>
<blockquote>
<div><p>The shrinkage constant in the interval [0,1] to regularize the covariance matrix towards the identity
matrix. This ensures that the covariance matrix is positive definite.
A larger xi value biases our estimate towards the supplied R matrix, a lower value biases it towards the
identity matrix.
If None, then xi is sampled from a beta distribution with shape parameters specified by the tuple
xi_prior_shape.</p>
</div></blockquote>
<p><strong>xi_prior_shape</strong> : tuple</p>
<blockquote>
<div><p>Shape parameters of the beta prior placed on the model parameter xi, specified as a 2-tuple of real values.
This argument is ignored and xi is not sampled, if it is specified explicitly using the xi parameter.</p>
</div></blockquote>
<p><strong>check_finite</strong> : bool</p>
<blockquote>
<div><p>Whether to check that the input matrices contain only finite numbers. Disabling may give a performance gain,
but may result in problems (crashes, non-termination) if the inputs do contain infinities or NaNs.
This parameter is passed on to several linear algebra functions in scipy internally.</p>
</div></blockquote>
<p><strong>min_eigenval</strong> : float</p>
<blockquote>
<div><p>Minimum Eigenvalue we can accept in the covariance matrix. Any eigenvalues encountered below this threshold
are set to zero, and the resulting covariance matrix normalized to give ones on the diagonal.</p>
</div></blockquote>
<p><strong>jitter</strong> : float</p>
<blockquote class="last">
<div><p>A small value to add to the diagonals of the covariance matrix to avoid conditioning issues.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
<p class="rubric">Methods</p>
<dl class="method">
<dt id="bss.models.probit.Probit.log_joint">
<code class="descname">log_joint</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#bss.models.probit.Probit.log_joint" title="Permalink to this definition">¶</a></dt>
<dd><p>The joint log-likelihood value of given model parameters.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">float</p>
<blockquote class="last">
<div><p>A scalar joint log-likelihood value of the entire model, given the model parameters and the true Y values
stored in this model object.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="bss.models.probit.Probit.log_marg_like">
<code class="descname">log_marg_like</code><span class="sig-paren">(</span><em>gamma</em>, <em>gamma0</em>, <em>lamb</em>, <em>nu</em><span class="sig-paren">)</span><a class="headerlink" href="#bss.models.probit.Probit.log_marg_like" title="Permalink to this definition">¶</a></dt>
<dd><p>The marginal likelihood log-value of given model parameters.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>gamma</strong> : ndarray</p>
<blockquote>
<div><p>A dx1 vector of probit values, one per SNP</p>
</div></blockquote>
<p><strong>gamma0</strong> : float</p>
<blockquote>
<div><p>The probit threshold, All SNPs with gamma &gt; gamma0 are ‘activated’.</p>
</div></blockquote>
<p><strong>lamb</strong> : float</p>
<blockquote>
<div><p>Inverse squared global scale parameter for the regression weights.</p>
</div></blockquote>
<p><strong>nu</strong> : float</p>
<blockquote>
<div><p>The residual precision of the model.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">float</p>
<blockquote class="last">
<div><p>A scalar log-likelihood value of the model, given the model parameters (and the true Y values stored in
this model object).</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="bss.models.probit.Probit.ppi_distribution">
<code class="descname">ppi_distribution</code><span class="sig-paren">(</span><em>gamma</em>, <em>gamma0</em>, <em>lamb</em><span class="sig-paren">)</span><a class="headerlink" href="#bss.models.probit.Probit.ppi_distribution" title="Permalink to this definition">¶</a></dt>
<dd><p>Normal distribution for the posterior probability of inclusion.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>gamma</strong> : ndarray</p>
<blockquote>
<div><p>A dx1 vector of probit values, one per SNP</p>
</div></blockquote>
<p><strong>gamma0</strong> : float</p>
<blockquote>
<div><p>The probit threshold, All SNPs with gamma &gt; gamma0 are ‘activated’.</p>
</div></blockquote>
<p><strong>lamb</strong> : float</p>
<blockquote>
<div><p>Inverse squared global scale parameter for the regression weights.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Mvn</p>
<blockquote class="last">
<div><p>A multivariate normal distribution object representing the posterior probability of inclusion</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
<p class="rubric">Notes</p>
<dl class="docutils">
<dt>We’re interested in the posterior probability of inclusion:</dt>
<dd><div class="first last math">
\[p(y|X,\gamma,\gamma_0,\nu,\lambda)\]</div>
</dd>
<dt>marginalizing out the effect size captured by <span class="math">\(\beta\)</span>:</dt>
<dd><div class="first last math">
\[\beta | \nu,\lambda,\Gamma \propto \mathcal{N}(0, (\nu\lambda)^{-1}\Gamma)\]</div>
</dd>
</dl>
<p>The degenerate Gaussian form of the <span class="math">\(\beta\)</span> prior above (note that the covariance matrix <span class="math">\(\Gamma\)</span>
is a diagonal matrix of indicator values) allows us to perform this marginalization in closed form:</p>
<blockquote>
<div><div class="math">
\[ \begin{align}\begin{aligned}p(y|X,\gamma,\gamma_0,\nu,\lambda)\\= \int \int \mathcal{N} (y|\beta_01_n + X\beta,\nu^{-1}I_n \;
\mathcal{N} (\beta|0, (\nu\lambda)^{-1}\Gamma)) \;
\mathcal{N}(\beta_0|0,(\nu\lambda)^{-1}) \; d\beta d\beta_0\\= \int \mathcal{N} (y|\beta_0 1_n, \nu^{-1}(\lambda^{-1}X\Gamma X^T + I_n)) \;
\mathcal{N}(\beta_0 | 0, (\nu\lambda)^{-1}) d\beta_0\\= \mathcal{N} (y|0, \nu^{-1} \lambda^{-1}(1_n1_n^T + X\Gamma X^T) + I_n))\end{aligned}\end{align} \]</div>
</div></blockquote>
</dd></dl>

<dl class="method">
<dt id="bss.models.probit.Probit.probit_distribution">
<code class="descname">probit_distribution</code><span class="sig-paren">(</span><em>xi</em><span class="sig-paren">)</span><a class="headerlink" href="#bss.models.probit.Probit.probit_distribution" title="Permalink to this definition">¶</a></dt>
<dd><p>The probit distribution of the model, driven by a latent Gaussian field.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>xi</strong> : float</p>
<blockquote>
<div><p>The shrinkage constant in the interval [0,1] to regularize the covariance matrix towards the identity
matrix. This ensures that the covariance matrix is positive definite.
A larger xi value biases our estimate towards the supplied R matrix, a lower value biases it towards the
identity matrix.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">The normal distribution of the probits (<cite>gamma</cite>) of the model.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="bss.models.probit.Probit.run_mcmc">
<code class="descname">run_mcmc</code><span class="sig-paren">(</span><em>iters=1000</em>, <em>burn_in=100</em>, <em>detailed=False</em><span class="sig-paren">)</span><a class="headerlink" href="#bss.models.probit.Probit.run_mcmc" title="Permalink to this definition">¶</a></dt>
<dd><p>Execute a run of MCMC on this model given the current state of model parameters.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>iters</strong> : int</p>
<blockquote>
<div><p>The total no. of iterations (excluding the burn-in) to run the MCMC simulations.</p>
</div></blockquote>
<p><strong>burn_in</strong> : int</p>
<blockquote>
<div><p>The total no. of burn-in iterations of the MCMC simulation.</p>
</div></blockquote>
<p><strong>detailed</strong> : bool, optional</p>
<blockquote>
<div><p>Whether to return a detailed trace of individual model parameters, or a trace of the joint log-likelihood
value.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">list, dictionary</p>
<blockquote class="last">
<div><p>A list of scalars (joint log-likelihood value of the entire model), or a dictionary of lists (with
keys <cite>inclusion, gamma0, lambda, nu, xi, joint, likelihood</cite>), and traces of individual model parameters as
values in the dictionary).</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="bss.models.probit.Probit.update_gamma">
<code class="descname">update_gamma</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#bss.models.probit.Probit.update_gamma" title="Permalink to this definition">¶</a></dt>
<dd><p>Apply MCMC transition operator to model parameter <span class="math">\(\gamma\)</span>. For updating <span class="math">\(\gamma\)</span>, we use
elliptical slice sampling (ESS) described in <a class="reference internal" href="index.html#murray2010" id="id2">[MAM10]</a></p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">On return, the <span class="math">\(\gamma\)</span> parameter of the model has been updated using a new sample.</td>
</tr>
</tbody>
</table>
<p class="rubric">Notes</p>
<p>ESS samples efficiently and robustly from latent Gaussian models when significant covariance structure is
imposed by the prior, as in the Gaussian processes and the present structured sparsity model.</p>
<p>ESS generates random elliptical loci using hte Gaussian prior and then searches along these loci to find
acceptable points for slice sampling. When the data ar weakly informative and the prior is strong, as is the
case here, the elliptical loci effectively captures the dependence between the variables and enable faster
mixing. Here, using ESS for <span class="math">\(\gamma\)</span> enables us to avoid directly sampling over the large discrete
space of sparsity patterns that makes unstructured spike-and-slab computationally challenging.</p>
</dd></dl>

<dl class="method">
<dt id="bss.models.probit.Probit.update_gamma0">
<code class="descname">update_gamma0</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#bss.models.probit.Probit.update_gamma0" title="Permalink to this definition">¶</a></dt>
<dd><p>Apply MCMC transition operator to the model parameter <span class="math">\(\gamma_0\)</span>, the sparsity threshold.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">On return, the <span class="math">\(\gamma_0\)</span> parameter of the model has been updated using a new sample.</td>
</tr>
</tbody>
</table>
<p class="rubric">Notes</p>
<p>The parameter <span class="math">\(\gamma_0\)</span> specifies the probit threshold and, conditioned on <span class="math">\(\gamma\)</span>, it determines
which entries on the diagonal of <span class="math">\(\Gamma\)</span> are zero and which are one. The conditional
density of <span class="math">\(\gamma_0\)</span> does not have a simple closed form, but can be efficiently sampled using the
exponential-expansion slice sampling algorithm described in <a class="reference internal" href="index.html#neal2003" id="id3">[Nea03]</a></p>
</dd></dl>

<dl class="method">
<dt id="bss.models.probit.Probit.update_lambda">
<code class="descname">update_lambda</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#bss.models.probit.Probit.update_lambda" title="Permalink to this definition">¶</a></dt>
<dd><p>Apply MCMC transition operator to model parameter <span class="math">\(\lambda\)</span>, the global weight inverse-scale parameter.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">On return, the <span class="math">\(\lambda\)</span> parameter of the model has been updated using a new sample.</td>
</tr>
</tbody>
</table>
<p class="rubric">Notes</p>
<p>The parameter <span class="math">\(\lambda\)</span> determines the scale of the “slab” portion of the weight prior. The conditional
density of <span class="math">\(\lambda\)</span> does not have a simple closed form, but can be efficiently sampled using the
exponential-expansion slice sampling algorithm described in <a class="reference internal" href="index.html#neal2003" id="id4">[Nea03]</a></p>
</dd></dl>

<dl class="method">
<dt id="bss.models.probit.Probit.update_nu">
<code class="descname">update_nu</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#bss.models.probit.Probit.update_nu" title="Permalink to this definition">¶</a></dt>
<dd><p>Apply MCMC transition operator to model parameter <span class="math">\(\nu\)</span>, the residual precision.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">On return, the <span class="math">\(\nu\)</span> parameter of the model has been updated using this analytical approach.</td>
</tr>
</tbody>
</table>
<p class="rubric">Notes</p>
<p>The scalar nu determines the precision of the residual Gaussian noise of the response variables.
With the choice of a conjugate gamma prior distribution, the conditional posterior is also gamma:</p>
<div class="math">
\[ \begin{align}\begin{aligned}p(\nu | y, X, \Gamma, \lambda) \propto
\mathcal{N}(y|0,\nu^{-1}(\lambda^{-1}(1_n1_n^T + X \Gamma X^T) + I_n)) \, Gam(\nu|a_\nu, b_\nu)\\= Gam(\nu | a_\nu^{(n)}, b_\nu^{(n)})\\a_\nu^{(n)} = a_\nu + \frac{N}{2}\\b_\nu^{(n)} = b_\nu + \frac{1}{2} y^T (\lambda^{-1} (1_n 1_n^T + X \Gamma X^T) + I_n)^{-1} y\end{aligned}\end{align} \]</div>
</dd></dl>

<dl class="method">
<dt id="bss.models.probit.Probit.update_parameters">
<code class="descname">update_parameters</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#bss.models.probit.Probit.update_parameters" title="Permalink to this definition">¶</a></dt>
<dd><p>Update all model parameters in a single MCMC step.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">On return, the gamma, gamma0, lambda, nu and optionally the <cite>xi</cite> model parameter have been updated.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="bss.models.probit.Probit.update_xi">
<code class="descname">update_xi</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#bss.models.probit.Probit.update_xi" title="Permalink to this definition">¶</a></dt>
<dd><p>Apply MCMC transition operator to the shrinkage factor <span class="math">\(\xi\)</span>, used to regularize the covariance matrix
towards the identity matrix.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">On return, the <span class="math">\(\xi\)</span> parameter of the model has been updated using a new sample.</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-bss.models">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-bss.models" title="Permalink to this headline">¶</a></h2>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="bss.samplers.html" class="btn btn-neutral float-right" title="bss.samplers package" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="bss.html" class="btn btn-neutral" title="bss package" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2018, Princeton University.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'0.0.2',
            LANGUAGE:'None',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>
      <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  <script type="text/javascript">
      jQuery(function () {
          
          SphinxRtdTheme.Navigation.enableSticky();
          
      });
  </script> 

</body>
</html>